src/SideNav/index.js

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright 2020 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+import React from 'react'
+import PropTypes from 'prop-types'
+import classNames from 'classnames'
+
+import '@spectrum-css/sidenav'
+
+import './index.css'
+
+const nav = (items, selectedKeys, disabledKeys, onSelectionChange) => {
+  return (
+    <ul className='spectrum-SideNav spectrum-SideNav--multiLevel'>
+      {items.map((item, index) => {
+        return (
+          <li
+            className={classNames([
+              'spectrum-SideNav-item',
+              { 'is-selected': selectedKeys.includes(item.name) },
+              { 'is-disabled': disabledKeys.includes(item.name) }
+            ])}
+            key={index}
+          >
+            <a
+              href='#'
```

Comment:
So in Gastby we would use the navigate helper function in the onSelectionChange handler ? I think it would be cleaner to have a way to pass down the url to the href here wdyt ?

Reply:
That's what I wanted some feedback on. Initially that is the way I had set it up but it looks like onSelectionChange is the way rsv3 will handle it. So I'm happy to change the click handler of the anchor to do the navigation but wanted to have the option of being closer to rsv3. WDYT?

Reply:
> wanted to have the option of being closer to rsv3
Since the SideNav is not published yet in rsv3 it's possible that the API will be different from what we currently know. So  I think we can take the freedom to add some API that is different from rsv3.

Reply:
kk, I've updated the PR with an onclick handler that does the navigation.

Reply:
can we implement conditional rendering? i.e. if the url path is external, render an `<a>` element, but if it's internal, render a gatsby `<Link>` component?

Reply:
@jcalcaben Then it looks like this and I wonder is that any cleaner?

```javascript
            {item.path &&
            !item.path.startsWith('http://') &&
            !item.path.startsWith('https://') ? (
              <Link to={item.path} className='spectrum-SideNav-itemLink'>
                {item.name}
              </Link>
            ) : (
              <a
                href='#'
                className='spectrum-SideNav-itemLink'
                onClick={(e) => {
                  e.preventDefault()
                  if (item.path) {
                    document.location.href = item.path
                  }
                }}
              >
                {item.icon ? item.icon : ''}
                {item.name}
              </a>
            )}
```

Reply:
We can extract all the non-presentational code outside the `return` statement to make it cleaner. I can make a small PR to illustrate this if you don't mind.

Reply:
@jcalcaben I'm an open book, suggest away.

Reply:
These are the changes I'm proposing: https://github.com/jcalcaben/parliament-ui-components/commit/94839ab53cac2061edeb2985b436f0de76df74ac

It moves a lot of the logic outside of the return statement and reduces the amount of indentation.

Reply:
oh wait, I modified `Nav` in my proposal instead of `SideNav`

Reply:
@jcalcaben Looks good to me. Much cleaner. Can you just add your commit to the `sidenav` branch and we can merge this thing?